projects/cds/src/merchandising/cms-components/merchandising-carousel/merchandising-carousel.component.ts

```diff
@@ -23,6 +23,8 @@ import { MerchandisingCarouselModel } from './model/index';
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class MerchandisingCarouselComponent {
+  protected lastEventModelId: string;
```

Comment:
In order to avoid having the "side-effects" in the stream, let's try to find an alternative approach of keeping the last event model ID.
Maybe you can leverage the [scan](https://rxjs.dev/api/operators/scan) operator which is capable of holding the values inside of the stream?

Reply:
Tried to use megeScan but it didn't prevent event duplication
            
filter((carouselIsVisible) => carouselIsVisible),
              mergeScan((modelId,_) =>{
                return this.merchandisingCarouselComponentService
                  .sendCarouselViewEvent(
                    modelId,
                    this.fetchProducts$
                  ).pipe(
                    switchMap(model=>model.id)
                    );
              },''),
              switchMapTo(of())